doc/source/getting_started/code/ODL Basics.py

```diff
@@ -0,0 +1,198 @@
+# can simply think of a `numpy.ndarray`.
+
+x = odl.vector([1, 2, 3])
+print(x)
+
+
+# In fact, the default implementation of a `vector` is just a thin wrapper
+# around `numpy.ndarray`. However, this back-end can be switched for
+# something else, e.g., a CUDA-based data container. On the surface it
+# will look exactly the same, and the differences will be entirely under
+# the hood. More on that later.
+#
+# Let's look at the representation (`repr`) of `x` to get some additional
+# information beyond the pretty-printing above:
+
+x  # or print(repr(x))
+
+
+# Hm, this is interesting. Obviously, there's more to this vector than
+# just its values. The line printed above suggests that it is an `element`
+# of something called `fn`. Let's see if we can just look at the first
+# part of that expression, of course prepending the package name and a
+# dot, a trick we know from NumPy:
+
+odl.fn(3, 'int')
+
+
+# Okay, the code works, so we can try to run the whole line in the
+# representation of `x`.
+
+odl.fn(3, 'int').element([1, 2, 3])
+
+
+odl.fn(3, 'int').element([1, 2, 3]) == x
+
+
+odl.fn(3, 'int').element([1, 2, 3]) is x
+
+
+# So we can recreate `x` with its `repr` code, but it's a new vector, not
+# identical to `x`. Of course, we should be able to create other vectors,
+# too, and add them, subtract, multiply and so on.
+
+y = odl.vector([4, 5, 6])
+
+
+x + y
+
+
+x - y
+
+
+x * y
+
+
+# So that works exactly as in NumPy, too. Let's go a step further and
+# check out what this first part `fn(3, 'int')` really means. We already
+# saw that the code runs and produces an output, so we assign it to a
+# variable and look at its properties.
+
+s = odl.fn(3, 'int')
+
+
+print(s)
+
+
+type(s)
+
+
+# So `s` is an object of type `NumpyFn`. That gives at least a hint that
+# it uses NumPy internally, but what is it? More experienced Python users
+# know that we can print the
+# [docstring](https://www.python.org/dev/peps/pep-0257/#what-is-a-
+# docstring) of any Python object to get more information. Docstrings are
+# strings stored in the object as `object.__doc__` attribute:
+
+print(s.__doc__)
+
+
+# This text actually gives quite a lot of information. It tells us that
+# our object `s` created by the command `odl.fn(3, 'int')`
+# - is a [vector space](https://en.wikipedia.org/wiki/Vector_space),
+# - contains vectors of a fixed length `n` (that's the "n" in `Fn`),
+# - is constructed over the <a
+# href="https://en.wikipedia.org/wiki/Field_(mathematics)">field</a> (the
+# "F" in `Fn`) of real or complex numbers (usually) and
+# - represents its elements as `NumpyFnVector` instances.
+#
+# The last part is easiest to check since we already have an element `x`
+# to test:
+
+type(x)
+
+
+# Indeed, the type is the same as the one mentioned in the docstring. We
+# should be able to probe also the other properties.
+
+s.field
+
+
+s.size  # the equivalent of "length", NumPy users will recognize it
```

Comment:
No, it is not equivalent. len gives the maximum value of the first index, size the total size, i.e. the product of the maximum lengths of all indices. I guess any reader would be fine, but lets try to be strict. Simply say "the number of elements in the space" or something.